saltstack/hortonworks/salt/dhcp/etc/dhcp/dhclient-enter-hooks

```diff
@@ -21,6 +21,40 @@ log() {
     fi
 }
 
+generate_resolv_conf() {
+    log "Cleaning /etc/resolv.conf"
+    echo "; generated by /etc/dhcp/dhclient-enter-hooks" > /etc/resolv.conf
+    
+    log "Generating configuration for /etc/resolv.conf"
+    if [ -n "${search}" ]; then
+        echo "search ${search}" >> /etc/resolv.conf
+    fi
+
+    if [ -n "${RES_OPTIONS}" ]; then
+        echo "options ${RES_OPTIONS}" >> /etc/resolv.conf
+    fi
+
+    echo "nameserver 127.0.0.1" >> /etc/resolv.conf
+
+    if [ ! -f /etc/hostname ]; then
+        log "Creating new hostname"
+        echo $new_host_name > /etc/hostname
+        hostname -F /etc/hostname
+    fi
+
+    JAVA_VERSION=$(java -version 2>&1 | grep -oP 'version [^0-9]?(1\.)?\K\d+' || true)
```

Comment:
while this could work it would be more simple solution to use salt templating to generate the script during image burning instead of running this check every time some dhcp event happens

Reply:
Its a bit more complicated if we change the default java version at provision time what the plan is as a part of CRB-2312

Reply:
okay, thanks for the clarification